src/teacher.py

```diff
@@ -15,7 +16,8 @@
 class Teacher:
     """The teacher can respond to queries of various types."""
 
-    def __init__(self) -> None:
+    def __init__(self, train_tasks: Sequence[Task]) -> None:
+        self._train_tasks = train_tasks
```

Comment:
it's funny how we've decided what the teacher gets in the init versus has to create for itself via `create_env` and the ensuing lines...

Reply:
(not sure if any action item required, but it is weird)

Reply:
I think it's okay. It's not "cheating" for the teacher to know the train tasks, vs knowing env.simulate, since the agent has train tasks as well.